### ISSUE TITLE
First letter of onboardingCommitMessage is converted to lowercase although "commitMessageLowerCase" is set to "never"

### ISSUE THREAD

**Issue by rarkins** [renovatebot/renovate#25662]
### Discussed in https://github.com/renovatebot/renovate/discussions/25625

<div type='discussions-op-text'>

<sup>Originally posted by **mkrill** November  7, 2023</sup>
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Gitlab

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

Hello, 
in our onboarding configuration in `config.js` file , we try to initialize our onboardingCommitMessage like so `DIGIHUB-123456 configure Renovate`.

However, although having configured `semanticCommits: "enabled"` and `commitMessageLowerCase: "never"` in `config.js`, the resulting commit message in the onboarding branch is `chore(deps): dIGIHUB-223277 configure Renovate`.  Thus, instead of starting with uppercade `D`, it starts with lowercase `d`. This breaks our push rule for commit messages.

I am not a typescript person, but the commit responsible for this bug seems to be https://github.com/renovatebot/renovate/commit/ed73d380e854ef9f3a4c2d5a2dd2787dbfecfc1c and more concrete the following  function https://github.com/renovatebot/renovate/blob/main/lib/workers/repository/model/commit-message.ts#L71

IMO, there shouldn't be any lowercase conversion, if `commitMessageLowerCase` is set to "never".

I will add a reproduction repository as soon as I find the time to setup a corresponding GitHub repo.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

https://github.com/renovate-reproductions/nvm3

**Comment by HonkingGoose** [comment]
Are we duplicating this older issue?

- #22346

**Comment by domibarton** [comment]
I can confirm this bug is still relevant in `40.18.1`. Onboarding commit messages are still in lower case, even if `commitMessageLowerCase` is set to `never`.

**Comment by ddbtrmatic** [comment]
> Are we duplicating this older issue?
> 
>     * [Onboarding commit message ignores lowercase char #22346](https://github.com/renovatebot/renovate/issues/22346)

I don't think so. In this issue the report is the subject part of the semantic commit message is converted to lower-case. The other report is the opposite, where the first character of the entire commit message is converted to upper-case.
